Compatibility issue due to Sphinx  v4.0.0

## Checklist

<!-- Please carefully go through this checklist and put an 'x' inside the brackets: '[x]'  -->

* [X] I have followed every step on the [compiling wiki page](https://github.com/polybar/polybar/wiki/Compiling) and
  installed all necessary dependencies.
* [X] My problem is not described on the [known issues page](https://github.com/polybar/polybar/wiki/Known-Issues)
* [X] I have searched for other open and closed [issues](https://github.com/polybar/polybar/issues?q=is%3Aissue) that
  may have already reported this problem.
* [X] I was able to reproduce this build issue in a clean build

## Build Process

**How did you build polybar?**

<!--
Put an 'x' inside the brackets ([x]) of the entry that applies and respond to the question inside the parenthesis
-->

* [X] By cloning this repository (What is the output of `git describe --tags`?)
* 3.5.5 (also tested olders versions)

<!-- List the exact commands you are using to build polybar: -->
```bash
./build.sh
```

## Build Log
<!--
Copy-paste all the terminal output produced while building polybar.
This MUST include the output of the `cmake`, `make`, and/or `build.sh` commands, if you used them.
-->
```
CMake Error at doc/cmake_install.cmake:50 (file):
  file INSTALL cannot find
  "/home/florian/programs/polybar/build/doc/man/polybar.1": No such file or
  directory.
Call Stack (most recent call first):
  cmake_install.cmake:51 (include)


make: *** [Makefile:83: install] Error 1
** Failed to install executables...
```

## Additional context
<!-- Add any other context that you think is necessary about the problem here. -->
Due to the new Sphinx 4.0.0 release `/home/florian/programs/polybar/build/doc/man/polybar.1` doesn't exist, instead `/home/florian/programs/polybar/build/doc/man/1/polybar.1` is generated. And therefore INSTALL cannot find it.

Downgrading Sphinx to 3.5.4 solved the issue.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,9 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Build
+- Support building documentation on sphinx 4.0 ([`#2424`](https://github.com/polybar/polybar/issues/2424))
+
 ## [3.5.5] - 2021-03-01
 ### Build
 - Support older python sphinx versions again ([`#2356`](https://github.com/polybar/polybar/issues/2356))

--- a/doc/conf.py
+++ b/doc/conf.py
@@ -185,6 +185,9 @@ man_pages = [
     ('man/polybar.5', 'polybar', 'configuration file for polybar(1)', [], 5)
 ]
 
+man_make_section_directory = False
+
+
 # -- Options for Texinfo output ----------------------------------------------
 
 # Grouping the document tree into Texinfo files. List of tuples

